ni_measurementlink_service/_datatypeinfo.py

```diff
@@ -0,0 +1,59 @@
+from typing import NamedTuple
+
+from google.protobuf import type_pb2
+
+from ni_measurementlink_service.measurement.info import DataType, TypeSpecialization
+
+
+class DataTypeInfo(NamedTuple):
+    """Class that represents the information for each of the DataType enum values.
+
+    Attributes
+    ----------
+        grpc_field_type (type_pb2.Field.Kind): Field.Kind associated with the DataType
+
+        repeated (bool): Whether the DataType is a repeated field
+
+        type_specialization (TypeSpecialization): Specific type when value_type
+        can have more than one use
```

Comment:
FYI, Sphinx uses type hints, so listing the attribute types in the class docstring is redundant.

Also FYI, you can write docstrings for class attributes: https://stackoverflow.com/questions/6060813/how-to-document-fields-and-properties-in-python